### PR TITLE
[CI:DOCS] docs/buildah-commit.md: tweak some wording, add a --rm example

### PR DESCRIPTION
**Pull request by rhatdan**
Tweak wording to avoid suggesting that we add the actual text
"<none>:<none>" as a name for images when a name isn't specified to
`buildah commit`.

Add an example that uses `--rm` to the list of examples.

Co-authored-by: Tom Sweeney <tsweeney@redhat.com>
Signed-off-by: Nalin Dahyabhai <nalin@redhat.com>
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by rhatdan** [comment]
Replaces: https://github.com/containers/buildah/pull/2499

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @nalind @giuseppe @vrothberg @ashley-cui @QiWang19 PTAL

File: docs/buildah-commit.md [hunk 1]
```diff
@@ -10,8 +10,9 @@ buildah\-commit - Create an image from a working container.
 Writes a new image using the specified container's read-write layer and if it
 is based on an image, the layers of that image.  If *image* does not begin
 with a registry name component, `localhost` will be added to the name.  If
-*image* is not provided, the values for the `REPOSITORY` and `TAG` values of
-the created image will each be set to `<none>`.
+*image* is not provided, the image will have no name.  When an image has no
+name, the `buildah images` command will display `<none>` in the REPOSITORY and
```
Comment (TomSweeneyRedHat) [inline docs/buildah-commit.md:14]: My pref is to have the labels backticked.
```suggestion
name, the `buildah images` command will display `<none>` in the `REPOSITORY` and
```

File: docs/buildah-commit.md [hunk 2]
```diff
@@ -10,8 +10,9 @@ buildah\-commit - Create an image from a working container.
...
+TAG columns.
```
Comment (TomSweeneyRedHat) [inline docs/buildah-commit.md:15]:
```suggestion
`TAG` columns.
```

**Comment by TomSweeneyRedHat** [comment]
@rhatdan lint isn't happy...

**Comment by rhatdan** [comment]
Tests are green
@TomSweeneyRedHat @nalind @giuseppe @vrothberg @ashley-cui @QiWang19 PTAL

**Comment by QiWang19** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
now to root bors on....

**Comment by QiWang19** [comment]
bors retry

**Comment by TomSweeneyRedHat** [comment]
I love bors, I love bors, I love bors
bors retry

**Comment by TomSweeneyRedHat** [comment]
Time to save the planet @rhatdan?

**Comment by nalind** [comment]
bors retry

**Comment by rhatdan** [comment]
bors likes @nalind

**Pull request merged**